apps/GROMACS/ARCHER2_2023.4_cpe15.md

```diff
@@ -0,0 +1,305 @@
+
+Download and unpack the source
+
+```bash
+wget http://ftp.gromacs.org/pub/gromacs/gromacs-2023.4.tar.gz
+tar -xvf gromacs-2023.4.tar.gz
+cd gromacs-2023.4
+export SRC=$(pwd)
+```
+
+Setup correct modules and build environment
+-------------------------------------------
+
+Setup the programming environment:
+
+```bash
+module load cpe/22.12
+module load cray-python/3.9.13.1
+module load cray-fftw/3.3.10.3
+module load cmake/3.21.3
+export LD_LIBRARY_PATH=$CRAY_LD_LIBRARY_PATH:$LD_LIBRARY_PATH
+```
+
+These options are needed to be set correctly for any of the following steps.
+
+Configure and build the parallel, single-precision build
+--------------------------------------------------------
+
+Create a build directory in the source tree
+
+```bash
+cd ${SRC}
+mkdir build_mpi
+cd build_mpi
+```
+
+Use CMake to configure the build and then build and install. Remember to set the install 
+prefix to somewhere you have permission to write to.
+
+```bash
+export CC=cc
+export CXX=CC
+export GMX_INSTALL=/work/y07/shared/apps/core/gromacs/2023.4
+
+cmake                                       \
+      -DGMX_MPI=ON                          \
+      -DGMX_OPENMP=ON                       \
+      -DGMX_GPU=OFF                         \
+      -DGMX_DOUBLE=OFF                      \
+      -DGMX_BUILD_OWN_FFTW=ON               \
+      -DCMAKE_INSTALL_PREFIX=${GMX_INSTALL} \
+      ../
+
+make -j 8
+make install
+```
+
+Configure and build the parallel, double-precision build
+--------------------------------------------------------
+
+Create a build directory in the source tree
+
+```bash
+cd ${SRC}
+mkdir build_double
+cd build_double
+```
+
+Use CMake to configure the build and then build and install. Remember to set the install 
+prefix to somewhere you have permission to write to.
+
+```bash
+export CC=cc
+export CXX=CC
+export GMX_INSTALL=/work/y07/shared/apps/core/gromacs/2023.4
+
+cmake                                       \
+      -DGMX_MPI=ON                          \
+      -DGMX_OPENMP=ON                       \
+      -DGMX_GPU=OFF                         \
+      -DGMX_DOUBLE=ON                       \
+      -DGMX_BUILD_OWN_FFTW=ON               \
+      -DCMAKE_INSTALL_PREFIX=${GMX_INSTALL} \
+      ../
+
+make -j 8
+make install
+```
+
+Configure and build the serial, single-precision build
+-------------------------------------------------------
+
+The serial build is required for the GROMACS analysis utilities.
+
+Create a build directory in the source tree
+
+```bash
+cd ${SRC}
+mkdir build_serial
+cd build_serial
+```
+
+Use CMake to configure the build and then build and install. Remember to set the install 
+prefix to somewhere you have permission to write to.
+
+```bash
+export CC=cc
+export CXX=CC
+export GMX_INSTALL=/work/y07/shared/apps/core/gromacs/2023.4
+
+cmake                                       \
+      -DGMX_MPI=OFF                         \
+      -DGMX_OPENMP=OFF                      \
```

Comment:
While none of the analysis tools use MPI, some can use OpenMP to speed things up, so it's not required nor optimal to disable OpenMP.

Reply:
Thanks for the explanation, I've modified the current module (2024.2) and newer versions will continue to use `-DGMX_OPENMP=ON` for the serial version.